crates/wdk-build/src/cargo_make.rs

```diff
@@ -492,6 +499,57 @@ pub fn setup_path() -> Result<(), ConfigError> {
     Ok(())
 }
 
+/// Adds the WDK version to the environment.
+///
+/// # Errors
+///
+/// This function returns a [`ConfigError::WDKContentRootDetectionError`] if the
+/// WDK content root directory could not be found.
+pub fn setup_wdk_version() -> Result<String, ConfigError> {
+    let Some(wdk_content_root) = detect_wdk_content_root() else {
+        return Err(ConfigError::WDKContentRootDetectionError);
+    };
+    let version = get_latest_windows_sdk_version(&wdk_content_root.join("Lib"))?;
+    prepend_to_semicolon_delimited_env_var(WDK_VERSION_ENV_VAR, &version);
+    forward_env_var_to_cargo_make(WDK_VERSION_ENV_VAR);
+    Ok(version)
+}
+
+/// Sets the `WDK_INFVERIF_SAMPLE_FLAG` environment variable to contain the
+/// appropriate flag for building samples.
+///
+/// # Errors
+///
+/// This function returns a [`ConfigError::WDKContentRootDetectionError`] if the
+/// WDK version is ill-formed (that is, it does not match the form
+/// 10.x.yyyyy.z).
```

Comment:
I think ill-formed detection should happen in the function that exposes the version variable to the build environment (before it exposes it)